src/System.Private.Xml/src/System/Xml/Core/XmlTextWriter.cs

```diff
@@ -345,8 +350,25 @@ public int Indentation
         // Gets or sets which character to use for indenting when Formatting is set to "Indented".
         public char IndentChar
         {
-            get { return _indentChar; }
-            set { _indentChar = value; }
+            get { return _indentChars[0]; }
+            set
+            {
+                if (value == DefaultIndentChar)
+                {
+                    _indentChars = s_defaultIndentChars;
+                    return;
+                }
+
+                if (ReferenceEquals(_indentChars, s_defaultIndentChars))
+                {
+                    _indentChars = new char[IndentArrayLength];
```

Comment:
Can we do `_indentChars  = new string(value, IndentArrayLength).ToCharArray();`? So that we can omit the `for` loop.